### PR TITLE
bugfix: fix gateway path when object endwith SlashSeparator

### PR DESCRIPTION
**Pull request by sjp00556**
link to: https://github.com/juicedata/juicefs/issues/1129

### CODE REVIEW

**Comment by SandyXSD** [comment]
@sjp00556 Thank you for contribution~ Please check and sign the CLA first.

File: pkg/gateway/gateway.go [hunk 1]
```diff
@@ -520,7 +520,7 @@ func (n *jfsObjects) GetObjectInfo(ctx context.Context, bucket, object string, o
 		return
 	}
 	if strings.HasSuffix(object, sep) && !fi.IsDir() {
-		err = jfsToObjectErr(ctx, os.ErrNotExist, bucket, object)
```
Comment (davies) [inline pkg/gateway/gateway.go:523]: Will this change any behavior?

File: pkg/gateway/gateway.go [hunk 2]
```diff
@@ -152,7 +152,7 @@ func (n *jfsObjects) path(p ...string) string {
 	if len(p) > 0 && p[0] == n.conf.Format.Name {
 		p = p[1:]
 	}
-	return sep + minio.PathJoin(p...)
```
Comment (davies) [inline pkg/gateway/gateway.go:155]: After we fixed the bug in fs.Mkdir(), is this still needed?

File: pkg/gateway/gateway.go (hunk 1 above)
Comment (sjp00556) [reply pkg/gateway/gateway.go:523]: os.ErrNotExist will give 500 http code InternalError, syscall.ENOENT give 404 code. i think 404 is better

File: pkg/gateway/gateway.go (hunk 2 above)
Comment (sjp00556) [reply pkg/gateway/gateway.go:155]: this change can fix path problem like this: "/path/b////" => "/path/b", but i think it doesn't metter

File: pkg/gateway/gateway.go (hunk 1 above)
Comment (davies) [reply pkg/gateway/gateway.go:523]: I see, can you send another PR for this one?

Comment (sjp00556) [reply pkg/gateway/gateway.go:523]: ok, i sent a new PR: https://github.com/juicedata/juicefs/pull/1158

**Pull request closed without merging**